### ISSUE TITLE
Videos won't load or download

### ISSUE THREAD

**Issue by Snelso91** [SongTube/SongTube-App#154]
Hi,

I've been trying to use the latest release (SongTube 6.4.2+800) with both the general APK and the arm64-release APK on my OnePlus 8 Pro running Oxygen OS 11.0.10.10.IN11BA (Android 11), but after downloading just 3 audio files from 3 videos yesterday, the app no longer loads any videos.

Instead, the home page/featured videos list loads when the app opens, but it never refreshes itself/is always the same videos.
![Screenshot_20220219-104230](https://user-images.githubusercontent.com/78712321/154795693-5bd805ec-32f8-454f-ba55-786d2f504d30.jpg)

If I tap on a video or manually enter a video URL from clipboard it just shows the title and loading icon but no information or option to download.
![Screenshot_20220219-104242](https://user-images.githubusercontent.com/78712321/154795705-8131c248-1380-45e9-a331-06b039fc128f.jpg)

This occurs on both WiFi and mobile broadband (4G), and occurs both with and without a VPN enabled.
I have also tried using the previous release (SongTube 6.4.1+600) as well as restarting my phone and reinstalling the latest release several times but none of these have fixed it. I have granted the general files permission and the all files permission every time I install but that doesn't help either.

All other YouTube downloaders/apps I've tried are able to load the videos perfectly fine, so it seems it's not to do with my connection but is to do with the songtube app instead.

**Comment by Artx-II** [comment]
Fixed

**Comment by Snelso91** [comment]
Just to check, how can I download/compile the code that includes this new fix/patch instead of the old version 6.4.2+800?

**Comment by Snelso91** [comment]
@Artx-II Just to check, how can I download/compile the code that includes this new fix/patch instead of the old version 6.4.2+800?

**Comment by Artx-II** [comment]
You only need to update your flutter packages, since the fix was done on newpipeextractor_dart plugin.

**Comment by Snelso91** [comment]
> You only need to update your flutter packages
How exactly can I do this given that I manually installed the downloaded APK from GitHub?
I'm not familiar with how flutter works sorry.

**Comment by Artx-II** [comment]
Oh I misunderstood your question.

You'd have to download and setup Flutter in your system then download the source code.

Once you have done the above steps you can just open a terminal on the root folder of the source code and run the command: flutter build apk

**Comment by Snelso91** [comment]
>open a terminal on the root folder of the source code and run the command: flutter build apk

I tried doing this after setting up flutter (flutter doctor reports no issues), but I get the following error:
```
C:\Users\User\Downloads\SongTube-App-development>flutter build apk
Because every version of flutter_html from git depends on html ^0.15.0 and every version of newpipeextractor_dart from git depends on html ^0.13.3, flutter_html from git is incompatible with newpipeextractor_dart from git.
So, because songtube depends on both newpipeextractor_dart from git and flutter_html from git, version solving failed.
```

Any idea how to solve this in order to build the apk?
Thanks

**Comment by Clashkid155** [comment]
The app is signed which causes the error since the keystore file isn't found, so on [line 61](https://github.com/SongTube/SongTube-App/blob/9b35be85ea2dcfb62bdd395cde3ca3ab804852a2/android/app/build.gradle#L61),

You can try commenting [Line 58-63](https://github.com/SongTube/SongTube-App/blob/9b35be85ea2dcfb62bdd395cde3ca3ab804852a2/android/app/build.gradle#L58-L63) and [Line 81](https://github.com/SongTube/SongTube-App/blob/9b35be85ea2dcfb62bdd395cde3ca3ab804852a2/android/app/build.gradle#L81)

**Comment by Clashkid155** [comment]
> Any idea how to solve this in order to build the apk? Thanks

The app is signed which causes the error since the keystore file isn't found, so on [line 61](https://github.com/SongTube/SongTube-App/blob/9b35be85ea2dcfb62bdd395cde3ca3ab804852a2/android/app/build.gradle#L61),

You can try commenting [Line 58-63](https://github.com/SongTube/SongTube-App/blob/9b35be85ea2dcfb62bdd395cde3ca3ab804852a2/android/app/build.gradle#L58-L63) and [Line 81](https://github.com/SongTube/SongTube-App/blob/9b35be85ea2dcfb62bdd395cde3ca3ab804852a2/android/app/build.gradle#L81)

**Comment by Snelso91** [comment]
> You can try commenting [Line 58-63](https://github.com/SongTube/SongTube-App/blob/9b35be85ea2dcfb62bdd395cde3ca3ab804852a2/android/app/build.gradle#L58-L63) and [Line 81](https://github.com/SongTube/SongTube-App/blob/9b35be85ea2dcfb62bdd395cde3ca3ab804852a2/android/app/build.gradle#L81)

I did this, and the app did compile as shown below:
```
C:\Users\User\Downloads\SongTube-App-development>flutter build apk
The plugins `ext_storage, flutter_ffmpeg, flutter_screen` use a deprecated version of the Android embedding.
To avoid unexpected runtime failures, or future build failures, try to see if these plugins support the Android V2 embedding. Otherwise, consider removing them since a
future release of Flutter will remove these deprecated APIs.
If you are plugin author, take a look at the docs for migrating the plugin to the V2 embedding: https://flutter.dev/go/android-plugin-migration.

Building without sound null safety
For more information see https://dart.dev/null-safety/unsound-null-safety

Checking the license for package Android SDK Build-Tools 29.0.2 in C:\Users\User\AppData\Local\Android\sdk\licenses
License for package Android SDK Build-Tools 29.0.2 accepted.
Preparing "Install Android SDK Build-Tools 29.0.2 (revision: 29.0.2)".
"Install Android SDK Build-Tools 29.0.2 (revision: 29.0.2)" ready.
Installing Android SDK Build-Tools 29.0.2 in C:\Users\User\AppData\Local\Android\sdk\build-tools\29.0.2
"Install Android SDK Build-Tools 29.0.2 (revision: 29.0.2)" complete.
"Install Android SDK Build-Tools 29.0.2 (revision: 29.0.2)" finished.
Checking the license for package Android SDK Platform 31 in C:\Users\User\AppData\Local\Android\sdk\licenses
License for package Android SDK Platform 31 accepted.
Preparing "Install Android SDK Platform 31 (revision: 1)".
"Install Android SDK Platform 31 (revision: 1)" ready.
Installing Android SDK Platform 31 in C:\Users\User\AppData\Local\Android\sdk\platforms\android-31
"Install Android SDK Platform 31 (revision: 1)" complete.
"Install Android SDK Platform 31 (revision: 1)" finished.
WARNING: [Processor] Library 'C:\Users\User\.gradle\caches\modules-2\files-2.1\androidx.media2\media2-session\1.1.3\27a17d921a5d48327be3c4d25bcea79def4d1f9d\media2-session-1.1.3.aar' contains references to both AndroidX and old support library. This seems like the library is partially migrated. Jetifier will try to rewrite the library anyway.
 Example of androidX reference: 'androidx/media2/session/MediaBrowser$Builder'
 Example of support library reference: 'android/support/v4/media/session/MediaSessionCompat$Token'
Note: C:\Users\User\AppData\Local\Pub\Cache\git\apk_installer-d16b3f2faa9d836c92598b1712ffcdccb755d8d6\android\src\main\java\com\artxdev\apk_installer\ApkInstallerPlugin.java uses or overrides a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
Note: C:\Users\User\AppData\Local\Pub\Cache\hosted\pub.dartlang.org\flutter_isolate-2.0.0\android\src\main\java\com\rmawatson\flutterisolate\FlutterIsolatePlugin.java uses or overrides a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
Note: C:\Users\User\AppData\Local\Pub\Cache\hosted\pub.dartlang.org\flutter_isolate-2.0.0\android\src\main\java\com\rmawatson\flutterisolate\FlutterIsolatePlugin.java uses unchecked or unsafe operations.
Note: Recompile with -Xlint:unchecked for details.
Note: Some input files use or override a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
Note: C:\Users\User\AppData\Local\Pub\Cache\git\audio_service-1-8e97a9770d4eabd6c6ec0f384d294d64a07f0af6\android\src\main\java\com\ryanheise\audioservice\AudioServicePlugin.java uses unchecked or unsafe operations.
Note: Recompile with -Xlint:unchecked for details.
Note: C:\Users\User\AppData\Local\Pub\Cache\hosted\pub.dartlang.org\device_info-2.0.3\android\src\main\java\io\flutter\plugins\deviceinfo\DeviceInfoPlugin.java uses unchecked or unsafe operations.
Note: Recompile with -Xlint:unchecked for details.
Note: C:\Users\User\AppData\Local\Pub\Cache\git\file_operations-51c5333722304b2306fccde92672be759b6e078a\android\src\main\java\com\artxdev\file_operations\FileOperationsPlugin.java uses or overrides a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
Note: Some input files use or override a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
Note: Some input files use or override a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
Note: Some input files use or override a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
Note: Some input files use unchecked or unsafe operations.
Note: Recompile with -Xlint:unchecked for details.
Note: C:\Users\User\AppData\Local\Pub\Cache\git\flutter_pip-ee21b3281de266c88776bca6b6fd8f465a562081\android\src\main\java\com\artxdev\flutter_pip\FlutterPipPlugin.java uses or overrides a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
Note: C:\Users\User\AppData\Local\Pub\Cache\hosted\pub.dartlang.org\flutter_screen-0.0.2\android\src\main\java\flutter\plugins\screen\screen\ScreenPlugin.java uses or overrides a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
Note: C:\Users\User\AppData\Local\Pub\Cache\hosted\pub.dartlang.org\flutter_share-2.0.0\android\src\main\java\com\example\fluttershare\FlutterSharePlugin.java uses or overrides a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
Note: Some input files use or override a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
Note: C:\Users\User\AppData\Local\Pub\Cache\hosted\pub.dartlang.org\path_provider_android-2.0.11\android\src\main\java\io\flutter\plugins\pathprovider\PathProviderPlugin.java uses unchecked or unsafe operations.
Note: Recompile with -Xlint:unchecked for details.
Note: C:\Users\User\AppData\Local\Pub\Cache\hosted\pub.dartlang.org\system_shortcuts-1.0.0\android\src\main\java\com\example\systemshortcuts\SystemShortcutsPlugin.java uses or overrides a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
Note: Some input files use or override a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
Note: C:\Users\User\AppData\Local\Pub\Cache\hosted\pub.dartlang.org\volume-1.0.1\android\src\main\java\com\example\volume\VolumePlugin.java uses or overrides a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
WARNING: [Processor] Library 'C:\Users\User\.gradle\caches\modules-2\files-2.1\org.robolectric\shadows-framework\4.3\150103d5732c432906f6130b734e7452855dd67b\shadows-framework-4.3.jar' contains references to both AndroidX and old support library. This seems like the library is partially migrated. Jetifier will try to rewrite the library anyway.
 Example of androidX reference: 'androidx/test/runner/lifecycle/Stage'
 Example of support library reference: 'android/support/annotation/NonNull'
Running Gradle task 'assembleRelease'...                         1514.6s
√  Built build\app\outputs\flutter-apk\app-release.apk (71.8MB).
```

However, once I copied the output file (app-release.apk) to my phone I simply got an error message saying "You can't install the app on your device". Despite this, now that an updated version has been compiled and released (SongTube 6.4.3+1000), I was able to download and install that and the error is now fixed.

**Comment by Clashkid155** [comment]
If you got a can't install error, it's probably because you already had SongTube installed.
